src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -1696,17 +1703,142 @@ private void NotifyValueChanged(TransactionOperationContext context, string type
+                                                 DatabaseRecordChange.Replication | 
+                                                 DatabaseRecordChange.ETL | 
+                                                 DatabaseRecordChange.Subscription | 
+                                                 DatabaseRecordChange.Configuration;
+
+        public static DatabaseRecordChange TopologyAffected = DatabaseRecordChange.Backup |
+                                                              DatabaseRecordChange.Replication |
+                                                              DatabaseRecordChange.ETL |
+                                                              DatabaseRecordChange.Subscription;
+
+        public ConcurrentDictionary<string, ConcurrentQueue<(long Index, DatabaseRecordChange Change)>> CommandsHolder =
+            new ConcurrentDictionary<string, ConcurrentQueue<(long Index, DatabaseRecordChange Change)>>();
+
         private void NotifyDatabaseAboutChanged(TransactionOperationContext context, string databaseName, long index, string type, DatabasesLandlord.ClusterDatabaseChangeType change)
         {
             context.Transaction.InnerTransaction.LowLevelTransaction.OnDispose += transaction =>
             {
                 if (transaction is LowLevelTransaction llt && llt.Committed)
                 {
+                    var queue = CommandsHolder.GetOrAdd(databaseName, new ConcurrentQueue<(long Index, DatabaseRecordChange Type)>());
+                    queue.Enqueue((index, GetChangeType(type)));
+
                     ExecuteAsyncTask(index, () => DatabaseChanged?.Invoke(this, (databaseName, index, type, change)));
                 }
             };
         }
 
+        private DatabaseRecordChange GetChangeType(string type)
+        {
+            switch (type)
+            {
+                case nameof(UpdateUnusedDatabaseIdsCommand):
+                case nameof(ClusterTransactionCommand):
+                case nameof(CleanUpClusterStateCommand):
+                case nameof(AddOrUpdateCompareExchangeBatchCommand):
+                case nameof(DeleteValueCommand):
+                case nameof(DeleteCertificateCollectionFromClusterCommand):
+                case nameof(DeleteMultipleValuesCommand):
+                case nameof(CleanCompareExchangeTombstonesCommand):
+                case nameof(IncrementClusterIdentityCommand):
+                case nameof(IncrementClusterIdentitiesBatchCommand):
+                case nameof(UpdateClusterIdentityCommand):
+                case nameof(AddOrUpdateCompareExchangeCommand):
+                case nameof(RemoveCompareExchangeCommand):
+                case nameof(UpdateSnmpDatabaseIndexesMappingCommand):
+
+                case nameof(UpdateExternalReplicationStateCommand):
+                case nameof(AcknowledgeSubscriptionBatchCommand):
+                case nameof(UpdateEtlProcessStateCommand):
```

Comment:
Why `UpdateEtlProcessStateCommand` is handled as `DatabaseRecordChange.None;` ? I think we should check if we need to reset ETL so I think  `EtlLoader.Reset` should be called then?

https://github.com/ravendb/ravendb/blob/832bdb96a6f29ae4159b270925ca74b23eb29df0/src/Raven.Server/Documents/ETL/EtlLoader.cs#L486-L491

Reply:
You're right